src/viam/sdk/resource/resource.cpp

```diff
@@ -38,5 +38,12 @@ ResourceName Resource::get_resource_name(std::string name) {
     return r;
 }
 
+void stop_resource_if_stoppable(std::shared_ptr<Resource> resource) {
```

Comment:
Interesting; I agree with @stuqdog that this was an oversight on my part, and I don't quite understand your counterargument @acmorrow. If a user calls `stop_all` on the robot service and provides `extra` arguments in their `StopAllRequest`, should we not honor them?

Reply:
It's a misunderstanding on my part. I had not noticed the code in `RobotService_::StopAll` where it de-structures a provided `extra` into per-named-resource sub-`extra`s and forwards them, so it didn't seem to me that there would ever be a caller. I agree with @stuqdog that you need the overload.

I'm somewhat surprised by that behavior, to be honest, but I see that python does it too. Is it a convention of ours that `extra` to entry points which operate on multiple resources always have this structure?

Reply:
> Is it a convention of ours that extra to entry points which operate on multiple resources always have this structure?

I think `StopAll` is one of the only cases where there can be a bunch of `extra` arguments mapping to different resources. See the [proto definitions](https://github.com/viamrobotics/api/blob/main/proto/viam/robot/v1/robot.proto#L276-L283): a `StopAllRequest` is really a repeated `StopExtraParameters` where `StopExtraParameters` is a resource name and a set of extra params.

Reply:
I'm not aware of it being an official convention beyond `StopAll` where I believe a similar pattern exists in Go, but I imagine we'd do something similar on other `{Verb}All` methods.

Reply:
OK it is starting to make more sense. The `extra` being examined in `StopAll` is not our usual `extra` where we say something like:
```
  // Additional arguments to the method
  google.protobuf.Struct extra = 99;
```
It's an honest-to-goodness repeated field of the `StopAllRequest` message with well-known semantics. That's all fine then. I was alarmed by what seemed to me like the "general purpose" `extra` being imbued with specific semantics. Perhaps it would have been better if that field in `StopAllRequest` had been named something other than `extra`, but that ship has long sailed.

Sorry for the confusion on this one.

Reply:
> Please add the overload per @stuqdog's original request.

Hmm maybe I misunderstood, but there _should_ be two overloads of `stop_if_stoppable` now (one is `inline`d).